napari/_qt/qt_main_window.py

```diff
@@ -824,7 +824,14 @@ def _add_menus(self):
         )
         self.main_menu.addMenu(self.view_menu)
         # plugin menu
-        self.plugins_menu = menus.PluginsMenu(self)
+        self.plugins_menu = build_qmodel_menu(
+            MenuId.MENUBAR_PLUGINS,
+            title=trans._('&Plugins'),
+            parent=self._qt_window,
+        )
+        self.plugins_menu.aboutToShow.connect(
+            lambda: self._update_menu_state('plugins_menu')
+        )
```

Comment:
I don`t like this lambda. Could we just add two line method instead of using lamblda 

```
def _update_plugin_menu_state(self):
    self._update_menu_state('plugins_menu')
```

Reply:
Yes can do, would need to be done for all the other menus as well, which should be done in separate PR.